Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandStoreBuyItem.java

```diff
@@ -83,6 +95,36 @@ public void run(Map<String,String> parameters) throws UserErrorMessage {
 		ds.beginTransaction();
 		try
 		{
+			// If quantity item, need to refetch in transaction, to make 
+			// sure enough quantity available to purchase.
+			if(quantity > 1)
+			{
+				item = ds.refetch(item);
+				quantity = (Long)item.getProperty("quantity");
+				if(quantity == null) quantity = 1L;
+				if(buyQuantity > quantity)
+					throw new UserErrorMessage("Seller no longer has enough inventory to purchase!");

```

Comment:
The error message is a little ambiguous. Maybe include the numbers (you wanted x but he only has y).